transform.js

```diff
@@ -62,119 +64,93 @@ function transform (filename, options) {
-    // modify value node to prefix, and push css for transform
-    if (node.type === 'CallExpression' &&
-    node.callee && node.callee.type === 'Identifier' &&
-    node.callee.name === mname) {
-      // determine path
-      // - check if module import
-      // - don't prefix by default if module import
-      // - check if local file
+    function extractImportNodes (node) {
+      if (node.type !== 'CallExpression') return
+      if (!node.callee || node.callee.type !== 'Identifier') return
+      if (node.callee.name !== mname) return
+      const isLocalFile = /^\.{0,2}\//.test(node.arguments[0].value)
+      const localOptions = { global: !isLocalFile }
       try {
         var resolvePath = cssResolve(node.arguments[0].value, {
-          basedir: opts.basedir
+          basedir: path.dirname(filename)
         })
       } catch (err) {
-        if (err.message.substring(0, 18) !== 'Cannot find module') {
-          throw err
-        }
+        return self.emit('error', err)
       }
-      const fnp = resolvePath ||
-        path.join(path.dirname(filename), node.arguments[0].value)
-      if (resolvePath) opts.global = true
-      else transformStream.emit('file', fnp)
-      try {
-        const fnCss = fs.readFileSync(fnp, 'utf8').trim()
-
-        // read optional arguments passed in to node
-        // e.g. { global: false }
-        if (node.arguments[1] && node.arguments[1].properties) {
-          const props = node.arguments[1].properties
-          props.forEach(function (prop) {
-            opts[prop.key.name] = prop.value.value
-          })
-        }
-
-        sheetify(fnCss, fnp, opts, function (tf) {
-          nodes.push([ tf, node ])
-        })
 
-        return
-      } catch (e) {
-        const errMsg = 'sheetify: ' + e.path + ' cannot be imported'
-        return transformStream.emit('error', errMsg)
+      const iOpts = node.arguments[1]
+        ? xtend(opts, localOptions, staticEval(node.arguments[1]))
+        : xtend(opts, localOptions)
+
+      const val = {
+        filename: resolvePath,
+        opts: iOpts,
+        node: node
       }
+
+      nodes.push(val)
+    }
+  }
+
+  // iterate over nodes, and apply sheetify transformation
+  // then replace the AST nodes with new values
+  // (obj, fn) -> null
+  function iterate (val, done) {
+    if (val.css) return handleCss(val)
+    fs.readFile(val.filename, 'utf8', function (err, css) {
+      if (err) return done(err)
+      val.css = css
+      handleCss(val)
+    })
+
+    function handleCss (val) {
+      sheetify(val.css, val.filename, val.opts, function (err, css, prefix) {
+        if (err) return done(err)
+        const str = [
+          "((require('insert-css')(" + JSON.stringify(css) + ')',
+          ' || true) && ' + JSON.stringify(prefix) + ')'
+        ].join('')
+
+        const lolSemicolon = (val.node.parent.type === 'VariableDeclarator')
+          ? ''
+          : ';'
+        val.node.update(lolSemicolon + str)
```

Comment:
I literally don't have the mental capacity to get this working :(

@ahdinosaur if you want to give it a go, perhaps as a different patch?